### PR TITLE
Fixes loading embedded solr certs when current thread has different classsloader

### PR DESCRIPTION
**Pull request by tgeens**

### CODE REVIEW

**Review by anghelutar — APPROVED** [review]

File: alfresco-solrapi-client/alfresco-solrapi-client-spring/src/main/java/eu/xenit/alfresco/solrapi/client/spring/SolrRequestFactory.java [hunk 1]
```diff
@@ -1,19 +1,24 @@
 package eu.xenit.alfresco.solrapi.client.spring;
 
 import java.io.FileInputStream;
```
Comment (driesdeproost) [inline alfresco-solrapi-client/alfresco-solrapi-client-spring/src/main/java/eu/xenit/alfresco/solrapi/client/spring/SolrRequestFactory.java]: this import can be removed now

**Review by driesdeproost — APPROVED** [review]

File: alfresco-solrapi-client/alfresco-solrapi-client-spring/src/main/java/eu/xenit/alfresco/solrapi/client/spring/SolrRequestFactory.java [hunk 2]
```diff
@@ -65,9 +69,35 @@ private static KeyStore loadKeyStore(String keystoreType, String file, char[] pa
...
+        try {
+            if (resource.startsWith("classpath:")) {
+                String path = resource.substring("classpath:".length());
+                ClassPathResource cpResource = new ClassPathResource(path, SolrRequestFactory.class.getClassLoader());
+                if (cpResource.exists()) {
+                    return cpResource.getURL();
+                }
+            }
+        } catch (IOException e) {
+
```
Comment (kerkhofsd) [inline alfresco-solrapi-client/alfresco-solrapi-client-spring/src/main/java/eu/xenit/alfresco/solrapi/client/spring/SolrRequestFactory.java]: `logger.warn()` something?

**Review by kerkhofsd — APPROVED** [review]

File: alfresco-solrapi-client/alfresco-solrapi-client-spring/src/main/java/eu/xenit/alfresco/solrapi/client/spring/SolrRequestFactory.java (hunk 1 above)
Comment (tgeens) [reply alfresco-solrapi-client/alfresco-solrapi-client-spring/src/main/java/eu/xenit/alfresco/solrapi/client/spring/SolrRequestFactory.java]: Fixed

File: alfresco-solrapi-client/alfresco-solrapi-client-spring/src/main/java/eu/xenit/alfresco/solrapi/client/spring/SolrRequestFactory.java (hunk 2 above)
Comment (tgeens) [reply alfresco-solrapi-client/alfresco-solrapi-client-spring/src/main/java/eu/xenit/alfresco/solrapi/client/spring/SolrRequestFactory.java]: moved catch closer to `.getUrl()` and rethrowing `UncheckedIOException` now

**Pull request merged**